bundler/src/main.rs

```diff
@@ -110,6 +113,10 @@ struct ServeArgs {
     /// The URL of the OpenTelemetry collector to send traces to
     #[arg(long, env = "BUNDLER_OTEL_COLLECTOR_URL")]
     otel_collector_url: Option<Url>,
+    /// Paths to any static data files that should be included in the bundle - can be globs
+    // validate that string can be converted to a glob pattern but store it in its string form
+    #[arg(long, env = "BUNDLER_STATIC_DATA", value_parser = |raw: &'_ str| Pattern::new(raw).map(|_| raw.to_string()))]
+    static_data: Vec<String>,
```

Comment:
Why do we store this in it's string form instead of as a `Pattern`?

Reply:
Because the [`glob`](https://docs.rs/glob/0.3.1/glob/fn.glob.html) function expects a string instead of a `Pattern` so we were converting it back anyway.  If you want to use the Pattern directly, you need to do the directory walking yourself.

Reply:
Huh, I really expected there to be a `glob` that took a compiled `Pattern`, but apparently not. Is there any chance we could keep the `Pattern` throughout the code and convert it back to a string when used in `glob`? Just so it's clear what it is

Reply:
It did until I just changed it in the latest commit. The main reason for changing it back was to avoid the excessive noise in the tracing. We could add `skip/fields` annotations to manually make them legible but I figured this was easier.

```
static_data=[Pattern { original: "../charts/bundler/static/*.json", tokens: [Char('.'), Char('.'), Char('/'), Char('c'), Char('h'), Char('a'), Char('r'), Char('t'), Char('s'), Char('/'), Char('b'), Char('u'), Char('n'), Char('d'), Char('l'), Char('e'), Char('r'), Char('/'), Char('s'), Char('t'), Char('a'), Char('t'), Char('i'), Char('c'), Char('/'), AnySequence, Char('.'), Char('j'), Char('s'), Char('o'), Char('n')], is_recursive: false }]
```

Could add a `StaticDataGlob(String)` newtype with a `TryFrom<&str>` to make it cleared what it is and to clear up the `value_parser` stuff?

Reply:
Ah yeah, hadn't considered that, that's horribly noisy. A newtype with `TryFrom<&str>` (or `FromStr`?) sounds like a call

Reply:
Went with `FromStr` because it's the only one clap handles by default.

Reply:
If I understand correctly, `From<&str>` should be for lossless conversions to other string-like types, whilst `FromStr` can be that or lossy parsing